### Fisher - Lady Tasting Tea

The original problem was to determine the probability of the lady correctly picking the 4 cups from 8 with milk in first, if it was through purely random selection. The probability is approx. 1.5 %


#### Exercise 1

This exercise is to calculate the minimum number of cups of tea required to ensure the probability of randomly selecting the correct cups is less than or equal to 1%.


The following code is adapted from - https://stackoverflow.com/a/4941846

In [4]:
# Importing Math Module

import math

If selecting 4 cups from 8 gives a 1.5% chance, then we need to increase the sample size to get a probability closer to 1%.

In [ ]:
# Selecting 4 cups from 9 - number of possible combinations
math.comb(9,4)

In [ ]:
# Probability of correctly selecting 4 from 9 -rounded to 2 places

round(1/126*100, 2)


From the above, we can see that increasing the sample size to 9 gives an approx. 0.79 % chance of randomly selecting the correct 4 cups. 

If we try 4 cups from 10, we can see the chances of selecting the correct cups further reduced.

In [ ]:
round(1/math.comb(10,4)*100, 2)

Selecting 4 cups from 10 gives an approx. 0.48% chance of randomly choosing the correct cups.

#### 1.1 - Bonus

How many would be required if you were to let the taster get one cup wrong while maintaining the 1% threshold?

If the tester is allowed to get one wrong, then using the original problem - they can pick 5 cups from 8.

In [5]:
round(1/math.comb(8,5)*100, 2)

1.79

This gives a 1.79 % chance, up from 1.5 %. We will increase the sample size again to 9 - so selecting 5 from 9 (4 correct, 1 wrong)

In [6]:
round(1/math.comb(9,5)*100, 2)

0.79

Again, this gives a 0.79% chance as selecting 5 from 9 is the statistically the same as selecting 4 from 9.